tooling/test/test.go

```diff
@@ -126,6 +128,10 @@ func Run(t *testing.T, tests SugarTests) {
 
 			// send request
 			log.Debugf("Querying %s", url)
+			timeout, cancel := context.WithTimeout(context.Background(), 2 * time.Minute)
+			defer cancel()
+			req = req.WithContext(timeout)
```

Comment:
Why is this necessary? IIRC there is a Go testcase timeout that one could configure to exit early. If we want to control it at this level too, I think it should be configurable.

Reply:
As far as I understand, there is only a single, global, go timeout: https://github.com/golang/go/issues/48157. That run hang'd on a single test for example: https://github.com/ipfs/gateway-conformance/actions/runs/4937755800/jobs/8826765779

> If we want to control it at this level too, I think it should be configurable.

I'd like to wait until we need that feature ([YAGNI](https://martinfowler.com/bliki/Yagni.html)). It's not hard to add but it'd make the code, configuration, and PR more complex without a use case right now.